tapalcatl_server/server.go

```diff
@@ -333,15 +331,11 @@ func main() {
 	// metrics writer configuration
 	var mw metricsWriter
 	if metricsStatsdAddr != "" {
-		if metricsInterval <= 0 {
-			logger.Fatalf("ERROR: metrics interval should be greater than 0")
-		}
 		udpAddr, err := net.ResolveUDPAddr("udp4", metricsStatsdAddr)
 		if err != nil {
 			logger.Fatalf("ERROR: Invalid metricsstatsdaddr %s: %s", metricsStatsdAddr, err)
 		}
-		metricsIntervalDuration := time.Second * time.Duration(metricsInterval)
-		mw = NewStatsdMetricsWriter(udpAddr, metricsStatsdPrefix, metricsIntervalDuration)
+		mw = NewStatsdMetricsWriter(udpAddr, metricsStatsdPrefix, logger)
```

Comment:
Was this concern just out of general curiosity? In this case the only reason I can see why the goroutine/channel would become garbage is if the handler went out of scope, and the reference in the main function disappeared, which to me implies the program itself is done. And the only blocking is on the goroutine that reads from the channel to process the requests. The request handler won't block; it'll just drop these stats if the channel is full and log it.